server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -373,8 +373,32 @@ public int hashCode() {
     }
 
     public static class Builder<KType, VType> {
+
+        // if the Builder was constructed from an existing ImmutableOpenMap, then this will be non-null (at least up until the
+        // point where the builder has been used to actually make some changes to the map that is being built -- see maybeInitializeMap)
+        private ImmutableOpenMap<KType, VType> original;
+
+        // if the Builder was constructed with a size (only), then this will be non-null
         private ObjectObjectHashMap<KType, VType> map;
 
+        // n.b. a constructor can either set original or it can set map, but it must not set both.
+
+        /**
+         * {@link Builder#maybeInitializeMap()} must be called before reading or writing via the {@code map} reference -- so every method
```

Comment:
It's strange to link to the method you're documenting here, let's just start the sentence with `This method` or something :)

Reply:
👍, fb8d8b9ac3a703759843e08c2765e1c546ac2377